v2/types.go

```diff
@@ -995,3 +995,59 @@ func ConvertBlockItems(input []*pb.BlockItem) []*BlockItem {
 
 	return result
 }
+
+// A chain epoch.
+type Epoch struct {
+	Value uint64
+}
+
+// Input to queries which take an epoch as a parameter.
+type isEpochRequest interface {
+	isEpochRequest()
```

Comment:
It's odd to have this function. Why do we need this interface with this function that does nothing?

Reply:
Apparently this is how one does interfaces in go, this is essentially a tagging interface. You don't explicictly implement an interface, you just implement every method the interface declares.. Something similar is done for `BlockHashInput`.